independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java

```diff
@@ -255,4 +260,54 @@ private String sanitizePathPrefix(String prefix) {
         return prefix;
     }
 
+    private static class MappersKey implements Comparable<MappersKey> {
+        private final String key;
+        private final URITemplate path;
+
+        public MappersKey(URITemplate path) {
+            this.path = path;
+
+            // create a key without any names. Names of e.g. default regex components can differ, but the component still has the same meaning.
+            StringBuilder keyBuilder = new StringBuilder();
+            for (URITemplate.TemplateComponent component : path.components) {
+                keyBuilder.append(component.type);
+                keyBuilder.append(";");
+                keyBuilder.append(component.literalText);
+                keyBuilder.append(";");
+                if (component.pattern != null) {
+                    // (?<id1>[a-zA-Z]+) -> [a-zA-Z]+
+                    String pattern = component.pattern.pattern();
+                    keyBuilder.append(component.pattern.pattern(), pattern.indexOf('>') + 1, pattern.length() - 1);
+                }
+                keyBuilder.append("|");
+            }
+
+            this.key = keyBuilder.toString();
```

Comment:
My guess is that it would be better if you could compute the hashCode here and cache it in a final field.